### PR TITLE
[fix](analysis) sorted partition when do sample analyze

### PR DESCRIPTION
**Pull request by freemandealer**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by freemandealer** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/statistics/OlapAnalysisTask.java [hunk 1]
```diff
@@ -255,7 +256,9 @@ protected Pair<List<Long>, Long> calcActualSampleTablets(boolean forPartitionCol
         List<Long> sampleTabletIds = new ArrayList<>();
         long actualSampledRowCount = 0;
         boolean enough = false;
-        for (Partition p : olapTable.getPartitions()) {
+        List<Partition> sortedPartitions = olapTable.getPartitions().stream().sorted(
+                Comparator.comparing(Partition::getName)).collect(Collectors.toList());
+        for (Partition p : sortedPartitions) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/OlapAnalysisTask.java:261]: do we need update inject stats of nereids tpcds shape test? @Jibing-Li

Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/OlapAnalysisTask.java:261]: Yes, we need to regenerate tpcds stats.

**Review by gavinchou — APPROVED** [review]

**Comment by freemandealer** [comment]
run performance

**Comment by freemandealer** [comment]
run buildall

**Comment by freemandealer** [comment]
run p0

File: fe/fe-core/src/main/java/org/apache/doris/statistics/OlapAnalysisTask.java (hunk 1 above)
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/OlapAnalysisTask.java:261]: Nereids tpcds shape test stats was collected using full analyze, so we don't need to update it.

**Review by Jibing-Li — APPROVED** [review]

**Pull request merged**